src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/on_vehicle_data_notification.cc

```diff
@@ -112,15 +112,23 @@ void OnVehicleDataNotification::Run() {
             mobile_api::FunctionID::OnVehicleDataID),
         appSO[idx].enumerate(),
         &params_permissions);
-
-    for (const auto& param : appSO[idx].enumerate()) {
-      const auto& allowed_params = params_permissions.allowed_params;
-      auto param_allowed = allowed_params.find(param);
-      if (allowed_params.end() == param_allowed) {
-        SDL_LOG_DEBUG("Param " << param << " is not allowed by policy for app "
-                               << notify_apps[idx]->app_id()
-                               << ". It will be ignored.");
-        appSO[idx].erase(param);
+    if (parameters_permissions_.allowed_params.empty() &&
+        parameters_permissions_.disallowed_params.empty() &&
+        parameters_permissions_.undefined_params.empty()) {
```

Comment:
in the else case all we check are allowed_params, why do we care if these other two arrays are empty here?

Reply:
This is the way that it is checked in `CommandImpl::CheckAllowedParameters`, I believe that all of the provided parameters would be in the `disallowed_params` list in the case that it is defined as `[]`, so these lists would not be empty.